https://lists.gnu.org/archive/html/bug-recutils/2024-10/msg00002.html

From 64708d4525603869e4ad22d468cb0d56e2f851c3 Mon Sep 17 00:00:00 2001
Message-ID: <64708d4525603869e4ad22d468cb0d56e2f851c3.1728191717.git.sam@gentoo.org>
In-Reply-To: <9f08064fef75ce02df9b6133d240c0fa84df458e.1728191717.git.sam@gentoo.org>
References: <9f08064fef75ce02df9b6133d240c0fa84df458e.1728191717.git.sam@gentoo.org>
From: Sam James <sam@gentoo.org>
Date: Sun, 6 Oct 2024 05:53:52 +0100
Subject: [PATCH 10/10] configure.ac: use pkg-config to find curl

A bunch of checks were already relying on pkg-config, let's do it
for curl too.

2024-10-06  Sam James <sam@gentoo.org>

	* configure.ac: Find curl via pkg-config.
---
 configure.ac | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index 7e47b9d..44bad76 100644
--- a/configure.ac
+++ b/configure.ac
@@ -89,14 +89,10 @@ AC_ARG_ENABLE([curl],
               [curl_enabled=$enableval], [curl_enabled=yes])
 have_curl=no
 AS_IF([test "x$curl_enabled" = "xyes"], [
-    AC_CHECK_LIB([curl], [curl_global_init], [have_curl=yes], [])
-    AS_IF([test "x$have_curl" = "xyes"], [
-        CURLLIBS=-lcurl
-    ], [
+    PKG_CHECK_MODULES([CURL], [libcurl], [have_curl=yes], [
         AC_MSG_ERROR([curl support enabled but libcurl not found])
     ])
 ], [])
-AC_SUBST([CURLLIBS])
 
 AC_ARG_ENABLE([uuid],
               AS_HELP_STRING([--enable-uuid],
-- 
2.46.2

